iceoryx_posh/test/integrationtests/test_popo_port_user_building_blocks.cpp

```diff
@@ -0,0 +1,357 @@
+
+    // subscriber port for single producer
+    SubscriberPortData m_subscriberPortDataSingleProducer{
+        TEST_SERVICE_DESCRIPTION, TEST_SUBSCRIBER_APP_NAME, VariantQueueTypes::SoFi_SingleProducerSingleConsumer};
+    SubscriberPortUser m_subscriberPortUserSingleProducer{&m_subscriberPortDataSingleProducer};
+    SubscriberPortSingleProducer m_subscriberPortRouDiSingleProducer{&m_subscriberPortDataSingleProducer};
+
+    // subscriber port for multi producer
+    SubscriberPortData m_subscriberPortDataMultiProducer{
+        TEST_SERVICE_DESCRIPTION, TEST_SUBSCRIBER_APP_NAME, VariantQueueTypes::SoFi_MultiProducerSingleConsumer};
+    SubscriberPortUser m_subscriberPortUserMultiProducer{&m_subscriberPortDataMultiProducer};
+    SubscriberPortMultiProducer m_subscriberPortRouDiMultiProducer{&m_subscriberPortDataMultiProducer};
+
+    // publisher port
+    vector<PublisherPortData, NUMBER_OF_PUBLISHERS> m_publisherPortDataVector;
+    vector<PublisherPortUser, NUMBER_OF_PUBLISHERS> m_publisherPortUserVector;
+    vector<PublisherPortRouDi, NUMBER_OF_PUBLISHERS> m_publisherPortRouDiVector;
+
+    inline CaproMessage waitForCaproMessage(const ConcurrentCaproMessageVector_t& concurrentCaproMessageVector,
+                                            const CaproMessageType& caproMessageType)
+    {
+        bool finished{false};
+        CaproMessage caproMessage;
+
+        // Wait until the expected CaPro message has arrived in the shared message vector between subscriber and the
+        // first publisher thread
+        do
+        {
+            // Add delay to allow other thread accessing the shared resource
+            std::this_thread::sleep_for(std::chrono::microseconds(10));
+            {
+                auto guardedVector = concurrentCaproMessageVector.GetScopeGuard();
+                if (guardedVector->size() != 0)
+                {
+                    caproMessage = guardedVector->back();
+
+                    if (caproMessage.m_type == caproMessageType)
+                    {
+                        guardedVector->pop_back();
+                        finished = true;
+                    }
+                }
+            }
+        } while (!finished);
+
+        return caproMessage;
+    }
+
+    template <typename SubscriberPortProducerType>
+    void subscriberThread(uint32_t numberOfPublishers,
+                          SubscriberPortProducerType& subscriberPortRouDi,
+                          SubscriberPortUser& subscriberPortUser)
+    {
+        bool finished{false};
+        optional<CaproMessage> maybeCaproMessage;
+        CaproMessage caproMessage;
+
+        // Wait for publisher to be ready
+        caproMessage = waitForCaproMessage(m_concurrentCaproMessageVector, CaproMessageType::OFFER);
+
+        // Subscribe to publisher
+        subscriberPortUser.subscribe();
+        maybeCaproMessage = subscriberPortRouDi.getCaProMessage();
```

Comment:
`auto maybeCaproMessage`? Then you save line 169 ;)

Reply:
Done.